### PR TITLE
En_Racedog OK

### PR DESCRIPTION
**Pull request by tom-overton**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

First off, huge shoutouts to Anon58 for doing the brunt of the work on func_80B25490; this function was driving me insane when I was trying to tackle it on my own.

Unfortunately, this actor is actually responsible for less than I thought it was when I first picked it up. This actor solely concerns the dogs that run around the track during the dog race; the dogs that you can see milling about before and after the race are actually En_Dg. Thus, similar to how I've been handling the Syateki series of actors, I'm leaving this actor mostly undocumented for now, but I *will* come back and document this, Aob1, and Dg at some point to understand how the race works.

### CODE REVIEW

**Review by AngheloAlf — APPROVED** [review]

File: src/overlays/actors/ovl_En_Racedog/z_en_racedog.c [hunk 1]
```diff
@@ -77,63 +138,531 @@ static DamageTable D_80B25ED0 = {
...
+    if (phi_v1 == 0) {
+        this->unk_288 = 0xC;
+    }
+}
+
+void func_80B258D8(EnRacedog* this, GlobalContext* globalCtx) {
+    Vec3s sp48 = gZeroVec3s;
+    s32 phi_v0 = (this->unk_290 == this->unk_292) ? 1 : 0;
+
+    if (phi_v0 != 0) {
```
[the remark this answers is not in the document]
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Racedog/z_en_racedog.c:587]: Yeah, inlining the condition inside the if doesn't match for some silly reason. But this does.

File: src/overlays/actors/ovl_En_Racedog/z_en_racedog.c [hunk 2]
```diff
@@ -77,63 +138,531 @@ static DamageTable D_80B25ED0 = {
...
+    func_80B255AC(this, globalCtx);
+}
+
+void func_80B24E14(EnRacedog* this) {
+    if (this->unk_290 % 2) {
+        D_80B25D88[this->unk_290].unk_0E =
+            (((gSaveContext.weekEventReg[42 + (this->unk_290 / 2)]) & 0xF0) >> 4) + 0x3539;
+    } else {
+        D_80B25D88[this->unk_290].unk_0E = ((gSaveContext.weekEventReg[42 + (this->unk_290 / 2)]) & 0xF) + 0x3539;
+    }
```
[the remark this answers is not in the document]
Comment (tom-overton) [reply src/overlays/actors/ovl_En_Racedog/z_en_racedog.c:346]: Yeah, Aob1 and Dg do it too.

Aob1:
```c
            idx = gSaveContext.weekEventReg[42 + idx2];
            gSaveContext.weekEventReg[42 + idx2] = idx | sp44[idx2];
```

Dg:
```c
 D_8098C2A8[this->unk_286].unk_04 =
                0x3538 + ((gSaveContext.weekEventReg[42 + (this->unk_286 / 2)] & (0x10 | 0x20 | 0x40 | 0x80)) >> 4);
```

The Dg one is identical to the Racedog one, so let me write Racedog's in that style.

File: src/overlays/actors/ovl_En_Racedog/z_en_racedog.c [hunk 3]
```diff
@@ -77,63 +141,532 @@ static DamageTable D_80B25ED0 = {
...
+
+s16 func_80B2478C(Path* path, s32 arg1, Vec3f* pos, f32* arg3) {
+    Vec3s* point;
+    f32 phi_f14;
+    f32 sp1C;
+    f32 xDiff;
+    f32 zDiff;
+
+    if (path != NULL) {
+        point = (Vec3s*)Lib_SegmentedToVirtual(path->points);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Racedog/z_en_racedog.c:197]:
```suggestion
        point = Lib_SegmentedToVirtual(path->points);
```

File: src/overlays/actors/ovl_En_Racedog/z_en_racedog.c [hunk 4]
```diff
@@ -77,63 +141,532 @@ static DamageTable D_80B25ED0 = {
...
+        zDiff = point->z - pos->z;
+    } else {
+        phi_f14 = 0.0f;
+        sp1C = 0.0f;
+        xDiff = 0.0f;
+        zDiff = 0.0f;
+    }
+
+    *arg3 = SQ(xDiff) + SQ(zDiff);
+    return Math_Acot2F(sp1C, phi_f14) * 10430.378f;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Racedog/z_en_racedog.c:211]:
```suggestion
    return RADF_TO_BINANG(Math_Acot2F(sp1C, phi_f14));
```

File: src/overlays/actors/ovl_En_Racedog/z_en_racedog.c [hunk 5]
```diff
@@ -77,63 +141,532 @@ static DamageTable D_80B25ED0 = {
...
+    zDistToBottomFront = this->actor.world.pos.z - arg1[1].y;
+    xDistToBottomBack = this->actor.world.pos.x - arg1[2].x;
+    zDistToBottomBack = this->actor.world.pos.z - arg1[2].y;
+    xDistToTopBack = this->actor.world.pos.x - arg1[3].x;
+    zDistToTopBack = this->actor.world.pos.z - arg1[3].y;
+
+    temp_f0 = ((xDistToTopFront * zDistToBottomFront) - (xDistToBottomFront * zDistToTopFront));
+    temp = (((xDistToBottomFront * zDistToBottomBack) - (xDistToBottomBack * zDistToBottomFront)));
+    if (temp_f0 * temp < 0.0f) {
+        return 0;
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Racedog/z_en_racedog.c:495]:
```suggestion
        return false;
```
Seems to return a boolean

**Review by hensldm — APPROVED** [review]
Just some minor things.

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**